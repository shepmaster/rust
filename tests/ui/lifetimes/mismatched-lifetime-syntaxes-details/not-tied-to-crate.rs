#![allow(mismatched_lifetime_syntaxes)]

#[warn(mismatched_lifetime_syntaxes)]
mod foo {
    fn bar(x: &'static u8) -> &u8 {
        //~^ WARNING lifetime flowing from input to output with different syntax
        x
    }

    #[deny(mismatched_lifetime_syntaxes)]
    fn baz(x: &'static u8) -> &u8 {
        //~^ ERROR lifetime flowing from input to output with different syntax
        x
    }
}

fn main() {}
