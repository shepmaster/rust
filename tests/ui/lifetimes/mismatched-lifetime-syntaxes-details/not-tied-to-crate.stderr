warning: lifetime flowing from input to output with different syntax
  --> $DIR/not-tied-to-crate.rs:5:16
   |
LL |     fn bar(x: &'static u8) -> &u8 {
   |                ^^^^^^^        --- the elided lifetime gets resolved as `'static`
   |                |
   |                this lifetime flows to the output
   |
note: the lint level is defined here
  --> $DIR/not-tied-to-crate.rs:3:8
   |
LL | #[warn(mismatched_lifetime_syntaxes)]
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: one option is to consistently use `'static`
   |
LL |     fn bar(x: &'static u8) -> &'static u8 {
   |                                +++++++

error: lifetime flowing from input to output with different syntax
  --> $DIR/not-tied-to-crate.rs:11:16
   |
LL |     fn baz(x: &'static u8) -> &u8 {
   |                ^^^^^^^        --- the elided lifetime gets resolved as `'static`
   |                |
   |                this lifetime flows to the output
   |
note: the lint level is defined here
  --> $DIR/not-tied-to-crate.rs:10:12
   |
LL |     #[deny(mismatched_lifetime_syntaxes)]
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: one option is to consistently use `'static`
   |
LL |     fn baz(x: &'static u8) -> &'static u8 {
   |                                +++++++

error: aborting due to 1 previous error; 1 warning emitted

